pkg/api/api.go

```diff
@@ -22,14 +23,21 @@ type PermitBaseFactsApi struct {
 
 type IPermitBaseFactsApi interface {
 	lazyLoadPermitContext(ctx context.Context, methodApiLevelArg ...config.APIKeyLevel) error
-	WaitForSync(timeout *time.Duration) *PermitBaseFactsApi
+	WaitForSync(timeout *time.Duration, policy ...config.FactsSyncTimeoutPolicy) *PermitBaseFactsApi
```

Comment:
Not sure this is a good way to sure I like this way. A better approach would be an interface that has an UpdateClientConfig function that accepts a client config and updates it.
With the current implementation if tomorrow we want to add another config for this, we would break the interface of WaitForSync

Reply:
Fixed